[dnf5] PreserveOrderMap: Add missing `const` in `const_reverse_iterator`, `iterator` to `cont_iterator`, add unit tests

jrohel opened the pull request:
* Add missing `const` in `const_reverse_iterator` type
* Added support for `iterator` to `const_iterator` conversion and `reverse_iterator` to `const_reverse_iterator` conversions. - Iterators should be implicitly convertible to their "const" version. Useful, for example, for the `erase` method, which accepts only `const_iterator`.
* Added unit tests

jrohel committed:
PreserveOrderMap: add missing `const` in `const_reverse_iterator`

jrohel committed:
PreserveOrderMap: Add unit tests

lukash commented on test/libdnf/preserve_order_map/test_preserve_order_map.cpp:
```diff
@@ -0,0 +1,194 @@
+void PreserveOrderMapTest::setUp() {}
+
+
+void PreserveOrderMapTest::tearDown() {}
```
You don't need to override these if they're empty?

jrohel replied:
It is true. I removed it.

lukash commented on test/libdnf/preserve_order_map/test_preserve_order_map.cpp:
```diff
@@ -0,0 +1,194 @@
+
+    // test at
+    CPPUNIT_ASSERT_EQUAL(std::string("4M"), map1.at(4));
+    CPPUNIT_ASSERT_THROW(map1.at(5), std::out_of_range);
```
It's not good to have a huge test that tests everything at once. You've already got the code split to blocks testing the various functionality. In general each block should roughly be one separate test. It can make sense to group some of them together to reduce the repetitive setup, but this is too much...

jrohel replied:
I split code and added more tests.

jrohel committed:
PreserveOrderMap: Added iterator to const_iterator conversion

Added support for `iterator` to `const_iterator` conversion and
`reverse_iterator` to `const_reverse_iterator` conversions.

Iterators should be implicitly convertible to their "const" version.

Useful, for example, for the `erase` method, which accepts only `const_iterator`.

lukash commented on include/libdnf/common/preserve_order_map.hpp:
```diff
@@ -51,6 +52,21 @@ class PreserveOrderMap {
 
+        // Allow iterator to const_iterator conversion
+        template <typename CItType = ContainerTypeIterator>
+        MyBidirIterator(
```
Looks good. FTR, somewhat unrelated, why is the class called `MyBidirIterator`? Could we come up with a better one? (I can merge as-is though)

jrohel committed:
PreserveOrderMap: Rename `MyBidirIterator` to `BidirIterator`
